src/test/docs.spec.ts

```diff
@@ -0,0 +1,55 @@
+import { createProjectSync, ts } from "@ts-morph/bootstrap"
+import * as fs from "fs"
+import * as path from "path"
+import { default as SimpleMarkdown } from "simple-markdown"
+
+describe("README examples", () => {
+  const readmePath = path.join(__dirname, "..", "..", "README.md")
+  const readmeContents = fs.readFileSync(readmePath)
+
+  // extract examples
+  const parser = SimpleMarkdown.parserFor(SimpleMarkdown.defaultRules)
+  const parsed = parser(readmeContents.toString())
+
+  let exampleNum = 1
+  parsed.forEach((block) => {
+    if (block.type !== "codeBlock" || block.lang != "typescript") return
+
+    describe(`compiles example ${exampleNum++}`, () => {
+      compile(block.content)
+    })
+  })
+})
+
+function compile(input: string): void {
+  const project = createProjectSync({ tsConfigFilePath: "tsconfig.json" })
+
+  // hack to make it possible to import the project source files instead of "@jetkit/cdk"
+  input = input.replace('from "@jetkit/cdk"', 'from "./src/index"')
```

Comment:
You can use the `resolutionHost` option to properly resolve the imports.

<https://github.com/dsherret/ts-morph/tree/latest/packages/bootstrap#custom-module-resolution>

Just pointing out, here it may be a 'overkill' to use. I think it's only worth to do that if we have many other imports or to avoid breaking the test if the formatting change from using double quotes to singles quotes)

Reply:
Yeah I was thinking that but would it mean that the project would have to be built before running the tests?

Reply:
I dont think so, the resolution should run during the compilation, at least looking in their example I think this would work:

```typescript
const project = await createProject({
    resolutionHost: (moduleResolutionHost, getCompilerOptions) => {
        return {
            resolveModuleNames: (moduleNames, containingFile) => {

                const compilerOptions = getCompilerOptions();
                const resolvedModules: ts.ResolvedModule[] = [];

                for (const moduleName of moduleNames) {
                     if(moduleName === "@jetkid/cdk") {
                        resolvedModules.push(ts.resolveModuleName(
                             './src/index',
                             containingFile,
                             compilerOptions,
                             moduleResolutionHost,
                        ).resolvedModule);
                     } else {
                        //just resolve normally.
                     }
                }
                return resolvedModules;
            },
        };
    },
});
```

I didn't tested it and just edited it in the github markdown.

Reply:
Actualy, maybe it's even easier, load a custom tsconfig file for the readmes including the source:

```typescript
import { Project } from "ts-morph";

const project = new Project({
    tsConfigFilePath: "path/to/tsconfig-for-md-tests.json",
});
```
-- tsconfig-for-md-tests.json
```json
{
  "compilerOptions": {
    "baseUrl": ".", // this must be specified if "paths" is specified.
    "paths": {
      "@jetkid/cdk": ["./src/index"]
    }
  }
}
```

https://ts-morph.com/setup/adding-source-files
https://www.staging-typescript.org/tsconfig#paths

Reply:
@yvdlima brilliant! Done in 2c76474